zerver/webhooks/jsonprinter/tests.py

```diff
@@ -0,0 +1,403 @@
+                }
+              ]
+            },
+            "type": "Error",
+            "mechanism": {
+              "synthetic": null,
+              "help_link": null,
+              "type": "generic",
+              "meta": null,
+              "handled": true,
+              "data": null,
+              "description": null
+            },
+            "value": "Something external broke."
+          }
+        ]
+      },
+      "fingerprint": [
+        "{{ default }}"
+      ],
+      "grouping_config": {
+        "enhancements": "eJybzDhxY3J-bm5-npWRgaGlroGxrpHxBABcTQcY",
+        "id": "newstyle:2019-10-29"
+      },
+      "hashes": [
+        "2058ed8b75cb570c9646adcd9dc21843"
+      ],
+      "key_id": "1146336",
+      "level": "error",
+      "location": null,
+      "logger": "",
+      "metadata": {
+        "type": "Error",
+        "filename": "/mnt/data/Documents/Stuff%20for%20Zulip/Repos/sentry/js/external.js",
+        "value": "Something external broke."
+      },
+      "received": 1592406695.060435,
+      "request": {
+        "cookies": null,
+        "url": "file:///mnt/data/Documents/Stuff%20for%20Zulip/Repos/sentry/js/trigger-exception-with-external.html",
+        "headers": [
+          [
+            "User-Agent",
+            "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:77.0) Gecko/20100101 Firefox/77.0"
+          ]
+        ],
+        "env": null,
+        "fragment": null,
+        "query_string": [],
+        "data": null,
+        "method": null,
+        "inferred_content_type": null
+      },
+      "sdk": {
+        "version": "5.17.0",
+        "name": "sentry.javascript.browser",
+        "packages": [
+          {
+            "version": "5.17.0",
+            "name": "npm:@sentry/browser"
+          }
+        ],
+        "integrations": [
+          "InboundFilters",
+          "FunctionToString",
+          "TryCatch",
+          "Breadcrumbs",
+          "GlobalHandlers",
+          "LinkedErrors",
+          "UserAgent"
+        ]
+      },
+      "timestamp": 1592404974.656,
+      "title": "Error: Something external broke.",
+      "type": "error",
+      "user": {
+        "geo": {
+          "city": "Hyderabad",
+          "region": "India",
+          "country_code": "IN"
+        },
+        "ip_address": "223.230.114.198"
+      },
+      "version": "7",
+      "url": "https://sentry.io/api/0/projects/hypro999-personal-organization/zulip-integration-testing/events/355c3b2a142046629dd410db2fdda003/",
+      "web_url": "https://sentry.io/organizations/hypro999-personal-organization/issues/1731239773/events/355c3b2a142046629dd410db2fdda003/",
+      "issue_url": "https://sentry.io/api/0/issues/1731239773/"
+    },
+    "triggered_rule": "Send a webhook payload to Webhook.site"
+  },
+  "installation": {
+    "uuid": "65027794-169a-4042-80d0-808ece377141"
+  },
+  "actor": {
+    "type": "application",
+    "id": "sentry",
+    "name": "Sentry"
+  }
+}
+```"""
```

Comment:
Can we avoid copying the logic here into the tests, and instead just check that the output equals the original file (or some simple transformation on it, like `json.dumps(json.load(...), indent_2)`?  

That would let us avoid a 400 line test file that will somewhat contribute to bloating the repository.

Also, could we make the fixture files symbolic links rather than copying them?

Reply:
@timabbott I have made the fixtures symlinks instead. Didn't realise this was possible in git repos. Now I do.
I have also changed the tests to access the original fixture instead.